pinot-query-planner/src/main/java/org/apache/pinot/query/planner/partitioning/KeySelector.java

```diff
@@ -37,4 +37,9 @@
   OUT getKey(IN input);
 
   int computeHash(IN input);
+
+  /**
+   * @return the hash-algorithm used for distributing rows
+   */
+  String hashAlgorithm();
```

Comment:
are we using this at all in this moment? if not i would suggest add this API in the future

Reply:
Yeah this is getting used because to make the colocation decision we need to know which hash algorithm is used to distribute the data in a given stage. In case a leaf stage decides to hash-distribute, this will set the ColocationKey of the receiving stage appropriately by setting the `ColocationKey._hashAlgorithm` to `KeySelector#hashAlgorithm()`.

Reply:
sounds good.